File: Projeto-Nosso-03/Project-Remider-Spring/src/main/java/br/com/fecaf/model/Lembrete.java
package br.com.fecaf.model;

import javax.persistence.*;
import java.time.LocalDate;
import java.time.LocalTime;
import java.util.List;

@Entity
@Table(name = "tbl_lembrete")
public class Lembrete {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String titulo;
    private LocalDate data;
    private LocalTime hora;
    private String prioridade;

    @Enumerated(EnumType.STRING)
    private TipoConteudo tipoConteudo;

    private String descricao;

    @OneToMany(mappedBy = "lembrete", cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.EAGER)
    private List<LembreteChecklist> checklist;


    @ElementCollection
    @CollectionTable(name = "lembrete_categorias", joinColumns = @JoinColumn(name = "lembrete_id"))
    @Column(name = "categoria")
    private List<String> categorias;

    private String cor;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User usuario;

    public enum TipoConteudo {
        DESCRICAO,
        CHECKLIST
    }

    @Column(nullable = false)
    private boolean concluido = false;

    // Getters e Setters

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public String getTitulo() { return titulo; }
    public void setTitulo(String titulo) { this.titulo = titulo; }

    public LocalDate getData() { return data; }
    public void setData(LocalDate data) { this.data = data; }

    public LocalTime getHora() { return hora; }
    public void setHora(LocalTime hora) { this.hora = hora; }

    public String getPrioridade() { return prioridade; }
    public void setPrioridade(String prioridade) { this.prioridade = prioridade; }

    public TipoConteudo getTipoConteudo() { return tipoConteudo; }
    public void setTipoConteudo(TipoConteudo tipoConteudo) { this.tipoConteudo = tipoConteudo; }

    public String getDescricao() { return descricao; }
    public void setDescricao(String descricao) { this.descricao = descricao; }

    public String getCor() { return cor; }
    public void setCor(String cor) { this.cor = cor; }

    public User getUsuario() { return usuario; }
    public void setUsuario(User usuario) { this.usuario = usuario; }

    public List<LembreteChecklist> getChecklist() {
        return checklist;
    }

    public void setChecklist(List<LembreteChecklist> checklist) {
        this.checklist = checklist;
    }

    public List<String> getCategorias() {
        return categorias;
    }

    public void setCategorias(List<String> categorias) {
        this.categorias = categorias;
    }

    public boolean isConcluido() {
        return concluido;
    }

    public void setConcluido(boolean concluido) {
        this.concluido = concluido;
    }
}

File: Projeto_Semestral_ADS-4NA_Lembraê-1.0/API/src/models/Lembrete.ts
import { 
  Entity,
  PrimaryGeneratedColumn,
  Column,
  ManyToOne,
  OneToMany,
  JoinColumn,
} from "typeorm";
import { User } from "./User";
import { Checklist } from "./Checklist";

@Entity({ name: "tbl_lembrete" })
export class Lembrete {
  @PrimaryGeneratedColumn()
  id!: number;

  @Column({ name: "titulo" })
  titulo!: string;

  @Column({ name: "descricao", type: "text", nullable: true })
  descricao?: string;

  @Column({ name: "data", type: "date", nullable: true })
  data?: string; // formato YYYY-MM-DD

  @Column({ name: "hora", type: "varchar", length: 5, nullable: true })
  hora?: string; // formato HH:MM

  @Column({ name: "prioridade", nullable: true })
  prioridade?: string;

  @Column({ name: "tipo_conteudo", nullable: true })
  tipo_conteudo?: string;

  get tipoConteudo() {
    return this.tipo_conteudo;
  }

  @Column({ name: "cor", nullable: true })
  cor?: string;

  @Column("simple-array", { name: "categorias", nullable: true })
  categorias?: string[]; // simples array compatível com front

  @Column({ name: "concluido", type: "tinyint", width: 1, default: 0 })
  concluido!: boolean;

  @ManyToOne(() => User, (user) => user.lembretes, {
    onDelete: "CASCADE",
  })
  @JoinColumn({ name: "user_id" })
  user!: User;

  @OneToMany(() => Checklist, (check) => check.lembrete, {
    cascade: true,
    eager: true,
    orphanedRowAction: "delete",
  })
  checklist!: Checklist[];
}